### PR TITLE
Use JTS Polygonizer to re-construct Polygons

### PR DESCRIPTION
**Pull request by matthieun**
### Description:

Use JTS Polygonizer to re-construct Polygons; this allows the construction of Polygons touching in one single point, which the legacy algorithm did not allow for.

In case the Polygonizer does not construct a polygon, we attempt a second pass with the legacy algorithm.

### Potential Impact:

More polygons are reconstructed. For example, the legacy algorithm can't build Austria's boundary because of that self touching point here: https://www.openstreetmap.org/node/2064631022. With the new algorithm, the boundary will have two outers which touch in that point.

### Unit Test Approach:

- Updated existing unit tests to account for JTS option
  - Added Polygon and Multipolygon `isSimilar()` using JTS's [HausdorffSimilarityMeasure](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/algorithm/match/HausdorffSimilarityMeasure.html) to allow for the inconsistencies in which the new algorithm starts the polygons; comparing WKTs in tests does not cut it anymore.
- Added specific unit tests for the enw capabilities

### Test Results:

Pass

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/geography/converters/MultiplePolyLineToPolygonsConverter.java [hunk 1]
```diff
@@ -31,25 +35,27 @@
     {
         private static final long serialVersionUID = -278028096455310936L;
 
+        public static final String OPEN_LOCATIONS_ARE = " Open Locations are: ";
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/converters/MultiplePolyLineToPolygonsConverter.java:38]: 👍 string constants 👍

File: src/test/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicingTest.java [hunk 2]
```diff
@@ -1030,11 +1030,13 @@ public void testMultiPolygonWithOverlappingSlicedInners()
     public void testMultiPolygonWithOverlappingUnslicedInners()
     {
         final Atlas rawAtlas = this.setup.getMultiPolygonWithOverlappingUnslicedInners();
+        System.out.println(rawAtlas.toStringDetailed());
```
Comment (adahn6) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicingTest.java:1033]: think this was left included by accident!

File: src/test/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicingTest.java [hunk 3]
```diff
@@ -1030,11 +1030,13 @@ public void testMultiPolygonWithOverlappingSlicedInners()
...
         final Atlas lbrSlicedAtlas = new RawAtlasSlicer(
                 AtlasLoadingOption.createOptionWithAllEnabled(boundary).setCountryCode("LBR"),
                 rawAtlas).slice();
         lbrSlicedAtlas.relations().forEach(relation ->
         {
+            System.out.println(relation);
```
Comment (adahn6) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicingTest.java:1039]: same here

**Review by adahn6 — DISMISSED** [review]
looks good overall, just a few minor changes.

File: src/main/java/org/openstreetmap/atlas/geography/converters/MultiplePolyLineToPolygonsConverter.java (hunk 1 above)
Comment (matthieun) [reply src/main/java/org/openstreetmap/atlas/geography/converters/MultiplePolyLineToPolygonsConverter.java:38]: Thank you sonar!

File: src/test/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicingTest.java (hunk 2 above)
Comment (matthieun) [reply src/test/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicingTest.java:1033]: Ooooh good catch!

**Review by lucaspcram — APPROVED** [review]

**Review by adahn6 — APPROVED** [review]

**Pull request merged**